src/kv/memstore.rs

```diff
@@ -73,37 +73,26 @@ impl Read for ReadTransaction<'_> {
 struct WriteTransaction<'a> {
     rt: ReadTransaction<'a>,
     pending: Mutex<HashMap<String, Vec<u8>>>,
-    open: bool,
 }
 
 impl WriteTransaction<'_> {
     fn new(store: &MemStore) -> WriteTransaction {
         return WriteTransaction {
             rt: ReadTransaction { store },
             pending: Mutex::new(HashMap::new()),
-            open: true,
         };
     }
-
-    fn check_open(&self) -> Result<()> {
-        match self.open {
-            true => Ok(()),
-            false => Err(StoreError::Str("Transaction already closed".into())),
-        }
-    }
 }
 
 #[async_trait(?Send)]
 impl Read for WriteTransaction<'_> {
     async fn has(&self, key: &str) -> Result<bool> {
-        self.check_open()?;
         match self.pending.lock().await.contains_key(key) {
             true => Ok(true),
             false => self.rt.has(key).await,
         }
     }
     async fn get(&self, key: &str) -> Result<Option<Vec<u8>>> {
-        self.check_open()?;
         match self.pending.lock().await.get(key) {
             Some(v) => Ok(Some(v.to_vec())),
```

Comment:
In the `Some` case it would definitely be possible to return a reference to the underlying vector.

In the `None` case, we end up copying the data out of the JS heap into our own vector.

In summary, maybe a TODO to eliminate the copy in both cases?

Reply:
This is true for MemStore, but not IdbStore, and these function signatures are defined by the traits. Given that, I don't think there's a good change to make here, even though you're right that the memory seems like it could be owned by the MemStore.

Even if we didn't have a trait though, wouldn't it be weird if you did a get(), then a put() over the same key, and the variable holding the get result had changed (or disappeared)? I don't see how you could define the memory to be owned by the store if it were to outlive the next write to the store without acrobatics.